source/library/CabalGild/Unstable/Main.hs

```diff
@@ -65,7 +71,14 @@ mainWith ::
 mainWith arguments = do
   flags <- Flag.fromArguments arguments
   config <- Config.fromFlags flags
-  context <- Context.fromConfig config
+  context <- do
+    context <- Context.fromConfig config
+    isTerm <- MonadHandle.hIsTerminalDevice $ Context.input context
+    if isTerm
+      then do
+        fs <- MonadDirectory.listDirectory "."
+        pure $ maybe id Context.setIO (Foldable.find ((== ".cabal") . FilePath.takeExtension) fs) context
```

Comment:
I think this should ensure there's exactly one `*.cabal` file. For normal packages, that should always be the case. But if you're in a directory with multiple package descriptions, Gild shouldn't just pick one of them arbitrarily.